src/controllers/loader/index.ts

```diff
@@ -0,0 +1,23 @@
+import { ResourceStoreContext } from '../../common/types';
+import { combine } from '../../common/utils/combine';
+import { entryPointsLoader } from '../../entry-points/loader';
+import { resourcesLoader } from '../../resources/loader';
+
+type Props = {
+  context: ResourceStoreContext | undefined;
+  resourceData: any;
+  timeout?: number;
+  isStatic?: boolean;
+};
+
+export const createCombinedLoader = (props: Props) => {
+  const { context, resourceData, isStatic } = props;
+  const entryPoint = entryPointsLoader();
+  const resources = resourcesLoader({
+    context,
+    resourceData,
+    isStatic,
+  });
+
+  return combine(entryPoint, resources);
```

Comment:
An alternative to this combine API (very redux-like) could be to just pass an array of loaders. I find it easier to digest imo.
It also means that Props are specific to each individual loader, instead of having `resourceData` passed even when entryPoint does not care about it

Reply:
Yeah, that should be fine, as it hides the `combine` as an implementation detail tbh

I'm not sure what you mean about the `resourceData` though, that should be the case with this API anyway.